### PR TITLE
[Doc] add notes on specialties on py-spy stack traces

### LINKED ISSUE #43129: [dashboard] py-spy flamegraph failures should have better docs
### What happened + What you expected to happen

In the dashboard one can trigger a flamegraph for a worker. This may not 100% work, e.g. 

1. If the process already exited
```
Failed to execute `['sudo', '-n', '/home/ray/anaconda3/bin/py-spy', 'dump', '-p', '2746']`.

=== stderr ===
Error: Failed to get process executable name. Check that the process is running.
Reason: No such file or directory (os error 2)
Reason: No such file or directory (os error 2)


=== stdout ===
```

2. if the job is sleeping
```
Failed to execute `['sudo', '-n', '/home/ray/anaconda3/bin/py-spy', 'record', '-o', PosixPath('/tmp/ray/session_2024-01-31_09-17-15_037017_341/logs/flamegraph_33695_cpu_profiling.svg'), '-p', '33695', '-d', '5', '-f', 'flamegraph']`.

=== stderr ===
[2024-01-31T22:37:27.932972718Z ERROR inferno::flamegraph] No stack counts found
Error: Failed to write flamegraph: I/O error: No stack counts found


=== stdout ===
py-spy> Sampling process 100 times a second for 5 seconds. Press Control-C to exit.
```

We know what's happening from py-spy, but a user may be clueless looking at them, e.g. when they click on a worker and got the "sleeping" result because it's waiting for a `ray.get`. 

We can do better by providing some Ray-provided words to instruct what's this error and what users should do. We can make it simple, that we don't parse the py-spy output and only give generic infos like:

```
- if you see "No such file or direction" your worker process may have been exited. 
- If you see "No stack counts found" your worker may be sleeping and did not have activity in the last 5s.
```

### Versions / Dependencies

master

### Reproduction script

-

### Issue Severity

Low: It annoys or frustrates me.

### PR DESCRIPTION
**Pull request by hongchaodeng**
## Related issue number

Closes #43129

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scottsun94** [comment]
Thanks for the fix!


@rynewang  Do you only want to add those notes to the documentation? Shall we also show them in the error message directly?

```
Failed to execute `['sudo', '-n', '/home/ray/anaconda3/bin/py-spy', 'record', '-o', PosixPath('/tmp/ray/session_2024-01-31_09-17-15_037017_341/logs/flamegraph_33695_cpu_profiling.svg'), '-p', '33695', '-d', '5', '-f', 'flamegraph']`.
- if you see "No such file or direction" in the stderr below, your worker process may have been exited. 
- If you see "No stack counts found" in the stderr below, your worker process may be sleeping and did not have activity in the last 5s.

=== stderr ===
[2024-01-31T22:37:27.932972718Z ERROR inferno::flamegraph] No stack counts found
Error: Failed to write flamegraph: I/O error: No stack counts found


=== stdout ===
py-spy> Sampling process 100 times a second for 5 seconds. Press Control-C to exit.
```

File: doc/source/ray-observability/user-guides/debug-apps/debug-hangs.rst [hunk 1]
```diff
@@ -23,6 +23,12 @@ trace is shown. To show native code frames, set the URL parameter ``native=1`` (
    * if you start Ray manually in a Docker container, follow the `py-spy documentation`_ to resolve it. 
    * if you are a KubeRay user, follow the :ref:`guide to configure KubeRay <kuberay-pyspy-integration>` and resolve it.
    
+.. note::
+   The following errors are conditional and not signals of failrues for your Python programs:
+
+   * If you see "No such file or direction", check if your worker process have exited.
+   * If you see "No stack counts found", check if your worker may be sleeping and not active in the last 5s.
```
Comment (scottsun94) [inline doc/source/ray-observability/user-guides/debug-apps/debug-hangs.rst:30]:
```suggestion
   * If you see "No stack counts found", check if your worker process was sleeping and not active in the last 5s.
```

**Review by rynewang — COMMENTED** [review]
LGTM. I wonder if we can also add it to the py-spy page, i.e. in https://github.com/ray-project/ray/blob/9a36755821a18179275257a456d5f8c811822464/dashboard/modules/reporter/profile_manager.py#L35

**Review by rynewang — APPROVED** [review]

File: doc/source/ray-observability/user-guides/debug-apps/debug-hangs.rst [hunk 2]
```diff
@@ -23,6 +23,12 @@ trace is shown. To show native code frames, set the URL parameter ``native=1`` (
    * if you start Ray manually in a Docker container, follow the `py-spy documentation`_ to resolve it. 
    * if you are a KubeRay user, follow the :ref:`guide to configure KubeRay <kuberay-pyspy-integration>` and resolve it.
    
+.. note::
+   The following errors are conditional and not signals of failrues for your Python programs:
```
Comment (angelinalg) [inline doc/source/ray-observability/user-guides/debug-apps/debug-hangs.rst:27]:
```suggestion
   The following errors are conditional and not signals of failures for your Python programs:
```

**Review by angelinalg — APPROVED** [review]
Just a typo. Consider installing Vale, a lint checker for your text. There's a VSCode extension, too! `go/vale`

**Pull request merged**